dev/com.ibm.ws.jaxrs.2.0.cdi.1.2_fat/fat/src/com/ibm/ws/jaxrs20/cdi12/fat/test/ContextandCDI12Test.java

```diff
@@ -0,0 +1,205 @@
+/*******************************************************************************
+ * Copyright (c) 2020 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.jaxrs20.cdi12.fat.test;
+
+import static org.junit.Assert.assertTrue;
+
+import java.util.List;
+
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import com.ibm.websphere.simplicity.ShrinkHelper;
+
+import componenttest.annotation.Server;
+import componenttest.custom.junit.runner.FATRunner;
+import componenttest.topology.impl.LibertyServer;
+
+@RunWith(FATRunner.class)
+public class ContextandCDI12Test extends AbstractTest {
+
+    public static final String[] ignore_messages =  new String[] { "CWWKW1001W" , "CWWKW1002W" , "CWWKE1102W", "CWWKE1106W" , "CWWKE1107W" };
+
+    @Server("com.ibm.ws.jaxrs20.cdi12.fat.contextandCDI")
+    public static LibertyServer server;
+
+    @BeforeClass
+    public static void setUp() throws Exception {
+        appname = "contextandCDI";
+        WebArchive app = ShrinkHelper.defaultDropinApp(server, appname, "com.ibm.ws.jaxrs20.cdi12.fat.contextandCDI");
+//        server.startServer();
+    }
+
+    @AfterClass
+    public static void tearDown() throws Exception {
+        server.stopServer(ignore_messages);
+    }
+
+    @Before
+    public void preTest() {
+        serverRef = server;
+    }
+
+    @After
+    public void afterTest() {
+        serverRef = null;
+    }
+    
+    
+    public void verifySuccess(String filterName, int messageSize) throws Exception {
+        String message = filterName + "#init: servletContext.getServletContextName contextandCDI";        
+        List<String> messages = serverRef.findStringsInLogs(message);
+        assertTrue("Expect to get CDI test message: " + message, messages.size() == messageSize);
+        message = filterName + "#filter#requestContext: servletContext.getServletContextName contextandCDI";        
+        messages = serverRef.findStringsInLogs(message);
+        assertTrue("Expect to get CDI test message: " + message, messages.size() == messageSize); 
+    }
+
+    @Test
+    public void testContextandCDIResource1() throws Exception {
+        if (server.isStarted()) {
+            server.stopServer(ignore_messages); 
+        }        
+        server.startServer(true);        
```

Comment:
These should be put in the `@Before` (to start the server) and `@After` (to stop it) methods - both methods might need to throw Exception.  Once in those methods, it will apply to all tests, so you can remove these lines from each test method.